src/app/clusters/scenes-server/scenes-server.cpp

```diff
@@ -0,0 +1,993 @@
+            sceneTable->GetAllSceneIdsInGroup(ctx.mCommandHandler.GetAccessingFabricIndex(), req.groupIdentifierFrom, sceneList)));
+
+        if (0 == capacity)
+        {
+            response.status = to_underlying(Protocols::InteractionModel::Status::ResourceExhausted);
+            ctx.mCommandHandler.AddResponse(mPath, response);
+            return;
+        }
+
+        for (auto & sceneId : sceneList)
+        {
+            SceneTableEntry scene(SceneStorageId(sceneId, req.groupIdentifierFrom));
+            //  Insert in table
+            ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+                ctx, response,
+                sceneTable->GetSceneTableEntry(ctx.mCommandHandler.GetAccessingFabricIndex(), scene.mStorageId, scene)));
+
+            scene.mStorageId = SceneStorageId(sceneId, req.groupIdentifierTo);
+
+            ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+                ctx, response, sceneTable->SetSceneTableEntry(ctx.mCommandHandler.GetAccessingFabricIndex(), scene)));
+        }
+
+        response.status = to_underlying(Protocols::InteractionModel::Status::Success);
+        ctx.mCommandHandler.AddResponse(mPath, response);
+        return;
+    }
+
+    SceneTableEntry scene(SceneStorageId(req.sceneIdentifierFrom, req.groupIdentifierFrom));
+    ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+        ctx, response, sceneTable->GetSceneTableEntry(ctx.mCommandHandler.GetAccessingFabricIndex(), scene.mStorageId, scene)));
+
+    scene.mStorageId = SceneStorageId(req.sceneIdentifierTo, req.groupIdentifierTo);
+
+    ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+        ctx, response, sceneTable->SetSceneTableEntry(ctx.mCommandHandler.GetAccessingFabricIndex(), scene)));
+
+    // Update Attributes
+    uint8_t scene_count = 0;
+    ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+        ctx, response, sceneTable->GetFabricSceneCount(ctx.mCommandHandler.GetAccessingFabricIndex(), scene_count)));
+
+    ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+        ctx, response, Attributes::SceneCount::Set(ctx.mRequestPath.mEndpointId, scene_count)));
+
+    ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+        ctx, response, sceneTable->GetRemainingCapacity(ctx.mCommandHandler.GetAccessingFabricIndex(), capacity)));
+
+    ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+        ctx, response, Attributes::RemainingCapacity::Set(ctx.mRequestPath.mEndpointId, capacity)));
+
+    Access::SubjectDescriptor descriptor = ctx.mCommandHandler.GetSubjectDescriptor();
+    if (AuthMode::kCase == descriptor.authMode)
+    {
+        ReturnOnFailure(AddResponseOnError<Commands::CopySceneResponse::Type>(
+            ctx, response, Attributes::LastConfiguredBy::Set(ctx.mRequestPath.mEndpointId, descriptor.subject)));
```

Comment:
Why can we skip doing that in the "copy all scenes" case?

Also, we have this code pasted enough times that it should really be a separate method.